packages/tailwindcss/src/theme.ts

```diff
@@ -89,9 +93,15 @@ export class Theme {
     return `--${this.prefix}-${key.slice(2)}`
   }
 
-  #clearNamespace(namespace: string) {
+  clearNamespace(namespace: string, clearOptions: ThemeOptions) {
```

Comment:
Made this public so we can selectively only clean `@theme default` values... See the test case as to why I think this is necessary.